app/src/features/rules/screens/rulesList/components/RulesList/components/RulesTable/hooks/useRuleTableColumns.tsx

```diff
@@ -43,6 +52,52 @@ const useRuleTableColumns = (options: Record<string, boolean>) => {
   } = useRulesActionContext();
   const isEditingEnabled = !(options && options.disableEditing);
 
+  const getRuleTypesDropdownOverlay = useMemo(
+    () => (groupId: string) => {
+      const checkIsPremiumRule = (ruleType: RuleType) => {
+        const featureName = `${ruleType.toLowerCase()}_rule`;
+        return !getFeatureLimitValue(featureName as FeatureLimitType);
+      };
+
+      return (
+        <Menu>
+          {Object.values(RULE_TYPES_CONFIG)
+            .filter((ruleConfig) => ruleConfig.ID !== 11)
+            .map(({ ID, TYPE, ICON, NAME }, index) => (
+              <div key={index}>
+                <PremiumFeature
+                  popoverPlacement="topLeft"
+                  onContinue={() => {
+                    createRuleAction(TYPE, SOURCE.RULE_GROUP);
+                    setSearchParams((params) => {
+                      params.set("groupId", groupId);
+                      return params;
+                    });
+                  }}
+                  features={[`${TYPE.toLowerCase()}_rule` as FeatureLimitType, FeatureLimitType.num_rules]}
+                  featureName={`${APP_CONSTANTS.RULE_TYPES_CONFIG[TYPE]?.NAME} rule`}
+                  source={SOURCE.RULE_GROUP}
+                >
+                  <Menu.Item icon={<ICON />} className="rule-selection-dropdown-btn-overlay-item">
+                    {NAME}
+                    {checkIsPremiumRule(TYPE) ? (
+                      <PremiumIcon
+                        key={index}
+                        placement="topLeft"
+                        source="rule_dropdown"
+                        featureType={`${TYPE.toLowerCase()}_rule` as FeatureLimitType}
+                      />
+                    ) : null}
+                  </Menu.Item>
+                </PremiumFeature>
+              </div>
+            ))}
+        </Menu>
+      );
+    },
+    [createRuleAction, getFeatureLimitValue]
```

Comment:
Abstract this as a common component so that we can use it in header and rule creation from group